src/live.rs

```diff
@@ -285,6 +311,22 @@ impl<'a> EmbedArea<'a> {
         Ok(())
     }
 
+    fn write_to_stdout(&mut self, buf: &[u8]) -> Result<()> {
+        self.file
+            .seek(SeekFrom::Start(0))
+            .chain_err(|| "seeking to start")?;
+        copy(&mut self.file.take(self.offset), &mut stdout())
+            .chain_err(|| "copying file to stdout")?;
+        stdout().write_all(buf).chain_err(|| "writing embed area")?;
+        let zeroes = vec![0; self.length - buf.len()];
+        stdout().write_all(&zeroes).chain_err(|| "writing zeros")?;
+        self.file
+            .seek(SeekFrom::Start(self.offset + self.length as u64))
+            .chain_err(|| "seeking to end of embed area")?;
+        copy(&mut self.file, &mut stdout()).chain_err(|| "copying file to stdout")?;
```

Comment:
I think, if you check `strace`, these copies will end up using a small I/O size.  It might make sense to wrap the input and output in `BufReader`/`BufWriter` with reasonably large buffers, to reduce syscall overhead.